### PR TITLE
Add clang.bazelrc setup from envoy.

### PR DESCRIPTION
**Pull request by incfly**
Signed-off-by: Jianfei Hu <hujianfei258@gmail.com>

#192 

FIPS boringssl requires clang to compile. This is the first step to setup the clang copmilation option.

### CODE REVIEW

File: bazel/setup_clang.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+echo "# Generated file, do not edit. If you want to disable clang, just delete this file.
+build:clang --action_env='PATH=${PATH}'
+build:clang --action_env=CC=clang
+build:clang --action_env=CXX=clang++
+build:clang --action_env='LLVM_CONFIG=${LLVM_PREFIX}/bin/llvm-config'
+build:clang --repo_env='LLVM_CONFIG=${LLVM_PREFIX}/bin/llvm-config'
+build:clang --linkopt='-L$(llvm-config --libdir)'
+build:clang --linkopt='-Wl,-rpath,$(llvm-config --libdir)'
+build:clang --define clang_cl=1
```
Comment (incfly) [inline bazel/setup_clang.sh:26]: This is the place that I diverts from the envoy upstream. the `clang_cl` is used in the boost config_settings option. Feel there could be a better way. @lizan

Comment (lizan) [reply bazel/setup_clang.sh:26]: clang_cl is the Windows option, do we use that?

Comment (incfly) [reply bazel/setup_clang.sh]: We don't. I only need a way to be able to defer that it's `--config=clang`. figured it out. Changed the BUILD.boost config_settings to use `value: {'action_env': 'CXX=clang++'}`

File: bazel/boost.bzl [hunk 2]
```diff
@@ -13,6 +13,10 @@ def boost_library(name, deps = []):
                     "libboost_{}.a".format(name),
                     "libboost_{}.dylib".format(name)
                 ],
+                "clang": [
```
Comment (Shikugawa) [inline bazel/boost.bzl:16]: Do we need it here? This seems the same as `//conditions:default`.

Comment (incfly) [reply bazel/boost.bzl:16]: there's a difference, it adds `toolset=clang`. without this, boost compilation fails, https://github.com/istio-ecosystem/authservice/issues/155

**Review by Shikugawa — APPROVED** [review]

**Pull request merged**